util/flagutil/parse.go

```diff
@@ -42,3 +42,17 @@ func ParseArgsWithValueModifier(command string, data interface{}, args []string,
 	}
 	return res, nil
 }
+
+// GetArgBestEffort returns the next element that follows argName
+func GetArgBestEffort(args []string, argName string) (string, bool) {
```

Comment:
This feels a bit hacky. I'm thinking of an edge case where the value (not the flag itself) of another flag is `--env-file` and this catches that incorrectly?

Reply:
yeah, `--build-arg --env-file +some-target` is such an edge case.

I can take a deeper look at `urfave/cli` to see if there's a different way we can parse once, then re-parse a second time once we pull out the `--env-file` value.

Reply:
It looks like the parsing is done via a private method: https://github.com/urfave/cli/blob/master/app.go#L241 which then calls https://github.com/urfave/cli/blob/e855c4cd1d5f87ae2a12c0e2620c0b747101c2bc/flag.go#L130-L140

I'll see if I can try a similar for loop in earthly where we call `Apply`.

Reply:
I'm wondering if it's actually not possible, without knowing the possible options first...

Reply:
It was easier than I thought, the only ugly part is we need to run newEarthlyApp twice:

```
for _, f := range newEarthlyApp(ctx, conslogging.ConsoleLogger{}).cliApp.Flags {
```

but fortunately `newEarthlyApp` is re-entrant.